### ISSUE TITLE
PaddleOCR dbnet 训练Hmean 达不到文档中的值

### ISSUE THREAD

**Issue by idontkonwher** [PaddlePaddle/Paddle#60601]
### bug描述 Describe the Bug

使用PaddleOCR (release 2.7)中的脚本，单卡训练模型：
 命令：python tools/train.py -c configs/det/det_mv3_db.yml
结果：
best metric, hmean: 0.7367632367632369, is_float16: False, precision: 0.7654385054488843, recall: 0.7101588830043332, fps: 82.37992130832617, best_epoch: 1080
文档中给出的精度：hmean为75.12

**Comment by idontkonwher** [comment]
测试了release 2.0 和 release 2.4，结果相近，无法达到75.12的hmean
paddle版本：2.4，
机器：A100 * 1
cuda：11.7

**Comment by tink2123** [comment]
repo里提供的配置文件是4卡训练的，单卡训练需要调整相应的学习率

**Comment by idontkonwher** [comment]
> repo里提供的配置文件是4卡训练的，单卡训练需要调整相应的学习率

感谢回复，我试试4卡训练

**Comment by YELLOWZAN** [comment]
> > repo里提供的配置文件是4卡训练的，单卡训练需要调整相应的学习率
> 
> 感谢回复，我试试4卡训练

四卡训练的模型，如果你单卡训练的话学习率需要除4

**Comment by idontkonwher** [comment]
@tink2123 @YELLOWZAN @jzhang533 我试了四卡运行：
python3 -m paddle.distributed.launch --log_dir=./log/ --gpus '0,1,2,3'  tools/train.py -c configs/det/det_mv3_db.yml
结果还是这样 best metric, hmean: 0.7353368880271448, precision: 0.7403611517813568, recall: 0.7303803562831006, fps: 50.36850404472764, best_epoch: 876

训练的配置如下：
[2024/01/08 15:02:25] root INFO: Architecture :
[2024/01/08 15:02:25] root INFO:     Backbone :
[2024/01/08 15:02:25] root INFO:         model_name : large
[2024/01/08 15:02:25] root INFO:         name : MobileNetV3
[2024/01/08 15:02:25] root INFO:         scale : 0.5
[2024/01/08 15:02:25] root INFO:     Head :
[2024/01/08 15:02:25] root INFO:         k : 50
[2024/01/08 15:02:25] root INFO:         name : DBHead
[2024/01/08 15:02:25] root INFO:     Neck :
[2024/01/08 15:02:25] root INFO:         name : DBFPN
[2024/01/08 15:02:25] root INFO:         out_channels : 256
[2024/01/08 15:02:25] root INFO:     Transform : None
[2024/01/08 15:02:25] root INFO:     algorithm : DB
[2024/01/08 15:02:25] root INFO:     model_type : det
[2024/01/08 15:02:25] root INFO: Eval :
[2024/01/08 15:02:25] root INFO:     dataset :
[2024/01/08 15:02:25] root INFO:         data_dir : ./train_data/icdar2015/text_localization/
[2024/01/08 15:02:25] root INFO:         label_file_list : ['./train_data/icdar2015/text_localization/test_icdar2015_label.txt']
[2024/01/08 15:02:25] root INFO:         name : SimpleDataSet
[2024/01/08 15:02:25] root INFO:         transforms :
[2024/01/08 15:02:25] root INFO:             DecodeImage :
[2024/01/08 15:02:25] root INFO:                 channel_first : False
[2024/01/08 15:02:25] root INFO:                 img_mode : BGR
[2024/01/08 15:02:25] root INFO:             DetLabelEncode : None
[2024/01/08 15:02:25] root INFO:             DetResizeForTest :
[2024/01/08 15:02:25] root INFO:                 image_shape : [736, 1280]
[2024/01/08 15:02:25] root INFO:             NormalizeImage :
[2024/01/08 15:02:25] root INFO:                 mean : [0.485, 0.456, 0.406]
[2024/01/08 15:02:25] root INFO:                 order : hwc
[2024/01/08 15:02:25] root INFO:                 scale : 1./255.
[2024/01/08 15:02:25] root INFO:                 std : [0.229, 0.224, 0.225]
[2024/01/08 15:02:25] root INFO:             ToCHWImage : None
[2024/01/08 15:02:25] root INFO:             KeepKeys :
[2024/01/08 15:02:25] root INFO:                 keep_keys : ['image', 'shape', 'polys', 'ignore_tags']
[2024/01/08 15:02:25] root INFO:     loader :
[2024/01/08 15:02:25] root INFO:         batch_size_per_card : 1
[2024/01/08 15:02:25] root INFO:         drop_last : False
[2024/01/08 15:02:25] root INFO:         num_workers : 8
[2024/01/08 15:02:25] root INFO:         shuffle : False
[2024/01/08 15:02:25] root INFO:         use_shared_memory : False
[2024/01/08 15:02:25] root INFO: Global :
[2024/01/08 15:02:25] root INFO:     cal_metric_during_train : False
[2024/01/08 15:02:25] root INFO:     checkpoints : None
[2024/01/08 15:02:25] root INFO:     debug : False
[2024/01/08 15:02:25] root INFO:     distributed : True
[2024/01/08 15:02:25] root INFO:     epoch_num : 1200
[2024/01/08 15:02:25] root INFO:     eval_batch_step : [0, 2000]
[2024/01/08 15:02:25] root INFO:     infer_img : doc/imgs_en/img_10.jpg
[2024/01/08 15:02:25] root INFO:     log_smooth_window : 20
[2024/01/08 15:02:25] root INFO:     pretrained_model : ./pretrain_models/MobileNetV3_large_x0_5_pretrained
[2024/01/08 15:02:25] root INFO:     print_batch_step : 10
[2024/01/08 15:02:25] root INFO:     save_epoch_step : 1200
[2024/01/08 15:02:25] root INFO:     save_inference_dir : None
[2024/01/08 15:02:25] root INFO:     save_model_dir : ./output/db_mv3/
[2024/01/08 15:02:25] root INFO:     save_res_path : ./output/det_db/predicts_db.txt
[2024/01/08 15:02:25] root INFO:     use_gpu : True
[2024/01/08 15:02:25] root INFO:     use_visualdl : False
[2024/01/08 15:02:25] root INFO: Loss :
[2024/01/08 15:02:25] root INFO:     alpha : 5
[2024/01/08 15:02:25] root INFO:     balance_loss : True
[2024/01/08 15:02:25] root INFO:     beta : 10
[2024/01/08 15:02:25] root INFO:     main_loss_type : DiceLoss
[2024/01/08 15:02:25] root INFO:     name : DBLoss
[2024/01/08 15:02:25] root INFO:     ohem_ratio : 3
[2024/01/08 15:02:25] root INFO: Metric :
[2024/01/08 15:02:25] root INFO:     main_indicator : hmean
[2024/01/08 15:02:25] root INFO:     name : DetMetric
[2024/01/08 15:02:25] root INFO: Optimizer :
[2024/01/08 15:02:25] root INFO:     beta1 : 0.9
[2024/01/08 15:02:25] root INFO:     beta2 : 0.999
[2024/01/08 15:02:25] root INFO:     lr :
[2024/01/08 15:02:25] root INFO:         learning_rate : 0.001
[2024/01/08 15:02:25] root INFO:     name : Adam
[2024/01/08 15:02:25] root INFO:     regularizer :
[2024/01/08 15:02:25] root INFO:         factor : 0
[2024/01/08 15:02:25] root INFO:         name : L2
[2024/01/08 15:02:25] root INFO: PostProcess :
[2024/01/08 15:02:25] root INFO:     box_thresh : 0.6
[2024/01/08 15:02:25] root INFO:     max_candidates : 1000
[2024/01/08 15:02:25] root INFO:     name : DBPostProcess
[2024/01/08 15:02:25] root INFO:     thresh : 0.3
[2024/01/08 15:02:25] root INFO:     unclip_ratio : 1.5
[2024/01/08 15:02:25] root INFO: Train :
[2024/01/08 15:02:25] root INFO:     dataset :
[2024/01/08 15:02:25] root INFO:         data_dir : ./train_data/icdar2015/text_localization/
[2024/01/08 15:02:25] root INFO:         label_file_list : ['./train_data/icdar2015/text_localization/train_icdar2015_label.txt']
[2024/01/08 15:02:25] root INFO:         name : SimpleDataSet
[2024/01/08 15:02:25] root INFO:         ratio_list : [1.0]
[2024/01/08 15:02:25] root INFO:         transforms :
[2024/01/08 15:02:25] root INFO:             DecodeImage :
[2024/01/08 15:02:25] root INFO:                 channel_first : False
[2024/01/08 15:02:25] root INFO:                 img_mode : BGR
[2024/01/08 15:02:25] root INFO:             DetLabelEncode : None
[2024/01/08 15:02:25] root INFO:             IaaAugment :
[2024/01/08 15:02:25] root INFO:                 augmenter_args :
[2024/01/08 15:02:25] root INFO:                     args :
[2024/01/08 15:02:25] root INFO:                         p : 0.5
[2024/01/08 15:02:25] root INFO:                     type : Fliplr
[2024/01/08 15:02:25] root INFO:                     args :
[2024/01/08 15:02:25] root INFO:                         rotate : [-10, 10]
[2024/01/08 15:02:25] root INFO:                     type : Affine
[2024/01/08 15:02:25] root INFO:                     args :
[2024/01/08 15:02:25] root INFO:                         size : [0.5, 3]
[2024/01/08 15:02:25] root INFO:                     type : Resize
[2024/01/08 15:02:25] root INFO:             EastRandomCropData :
[2024/01/08 15:02:25] root INFO:                 keep_ratio : True
[2024/01/08 15:02:25] root INFO:                 max_tries : 50
[2024/01/08 15:02:25] root INFO:                 size : [640, 640]
[2024/01/08 15:02:25] root INFO:             MakeBorderMap :
[2024/01/08 15:02:25] root INFO:                 shrink_ratio : 0.4
[2024/01/08 15:02:25] root INFO:                 thresh_max : 0.7
[2024/01/08 15:02:25] root INFO:                 thresh_min : 0.3
[2024/01/08 15:02:25] root INFO:             MakeShrinkMap :
[2024/01/08 15:02:25] root INFO:                 min_text_size : 8
[2024/01/08 15:02:25] root INFO:                 shrink_ratio : 0.4
[2024/01/08 15:02:25] root INFO:             NormalizeImage :
[2024/01/08 15:02:25] root INFO:                 mean : [0.485, 0.456, 0.406]
[2024/01/08 15:02:25] root INFO:                 order : hwc
[2024/01/08 15:02:25] root INFO:                 scale : 1./255.
[2024/01/08 15:02:25] root INFO:                 std : [0.229, 0.224, 0.225]
[2024/01/08 15:02:25] root INFO:             ToCHWImage : None
[2024/01/08 15:02:25] root INFO:             KeepKeys :
[2024/01/08 15:02:25] root INFO:                 keep_keys : ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask']
[2024/01/08 15:02:25] root INFO:     loader :
[2024/01/08 15:02:25] root INFO:         batch_size_per_card : 16
[2024/01/08 15:02:25] root INFO:         drop_last : False
[2024/01/08 15:02:25] root INFO:         num_workers : 8
[2024/01/08 15:02:25] root INFO:         shuffle : True
[2024/01/08 15:02:25] root INFO:         use_shared_memory : False
[2024/01/08 15:02:25] root INFO: profiler_options : None
[2024/01/08 15:02:25] root INFO: train with paddle 2.4.1 and device Place(gpu:0)
[2024/01/08 15:02:31] root INFO: Initialize indexs of datasets:['./train_data/icdar2015/text_localization/train_icdar2015_label.txt']
[2024/01/08 15:02:31] root INFO: Initialize indexs of datasets:['./train_data/icdar2015/text_localization/test_icdar2015_label.txt']

**Comment by tink2123** [comment]
可以参考官方的训练日志，对比下收敛趋势是否一致
[train.log](https://github.com/PaddlePaddle/Paddle/files/13884812/train.log)

**Comment by idontkonwher** [comment]
> 可以参考官方的训练日志，对比下收敛趋势是否一致 [train.log](https://github.com/PaddlePaddle/Paddle/files/13884812/train.log)

再次感谢回复，大致对比了一下log，loss的趋势是一致的，可能有visualdl log对比起来更方便些，最终的结果差异主要是precision差距较大，recall基本一致。对比log时发现了以下差异：
![image](https://github.com/PaddlePaddle/Paddle/assets/33867371/d6902080-6a16-479d-b3dc-19109a301e4c)
官方log使用的ratio_list为0.5，看文档这个是取数据时的比例，请问这个值是否会和batch size参数叠加从而影响实际运行时的batch size。

**Comment by tink2123** [comment]
这个值不会影响batch_size，但会影响网络训练的数据量。 可以先尝试修改下这个参数看看

**Comment by idontkonwher** [comment]
> 这个值不会影响batch_size，但会影响网络训练的数据量。 可以先尝试修改下这个参数看看

修改后会出现训练数据量减半的情况，从官方的log看每个epoch都会有1000张左右的samples，看起来应该是单卡训练的？我这多卡训练的话log上显示的samples只有官方log的1/4
![image](https://github.com/PaddlePaddle/Paddle/assets/33867371/3eb79e37-bc5a-4b03-b34d-58695fd8844a)
所以从log看我这边行为和官方log一致的是：单卡训练且ratio_list取1时 的结果，但是之前试过这种情况下也没法收敛到给出的hmean

**Comment by idontkonwher** [comment]
[
[train.log](https://github.com/PaddlePaddle/Paddle/files/13886495/train.log)
](url)
这是我的训练log，包含了两部分，第一部分时单卡的，第二部分是四卡的。

**Comment by idontkonwher** [comment]
help~! @tink2123

**Comment by tink2123** [comment]
我训练了一下，也没有达到75的hmean。这个模型很早了，可能在代码迭代的过程中引入了diff，我尝试一下去找能复现精度的版本。

**Comment by idontkonwher** [comment]
> 我训练了一下，也没有达到75的hmean。这个模型很早了，可能在代码迭代的过程中引入了diff，我尝试一下去找能复现精度的版本。

辛苦了，感谢大佬！

**Comment by idontkonwher** [comment]
hello，大佬~， Any progress？ @tink2123
